enhancements/network/multus-services.md

```diff
@@ -0,0 +1,239 @@
+
+#### Cluster IP for Multus
+
+When the service is created with `type: ClusterIP`, then Kubernetes assign cluster virtual IP for the services. User
+can access the services with given virtual IP. The request to virtual IP is automatically replaced with actual
+pods' network interface IP and send to the target services. User needs to make sure reachability to the target
+network, otherwise the request packet will be dropped.
+
+### User Stories
+
+- As an OpenShift developer, I require the ability to segregate my Kubernetes service data plane traffic onto different networks for reasons of security and performance.
+- As an OpenShift developer, I require the ability to segregate my Kubernetes service data plane traffic to a secondary interface that is associated with a different CNI plugin to provide functionality not available in the primary (default network) CNI plugin.
+- As an OpenShift networking administrator, I require the ability to provide access to an internally-facing corporate network and an externally-facing DMZ network, so application developers can choose which network (pod interface) to use for the different types of traffic that make up their application (e.g. web server:external, DB:internal).
+- As an OpenShift network administrator, I require the ability to use one NIC on my hosts for control plane traffic, and additional NICs for data plane traffic.
+
+### Implementation Details/Notes/Constraints
+
+In this enhancement, we will introduce following components (component names may be changed during development/community discussion):
+
+- Multus Service Controller
+- Multus Proxy
+
+The Multus service controller watches all pods' secondary interfaces, services and network attachment definitions and
+it creates `EndpointSlice` for the service. `EndpointSlice` contains secondary network IP address whidch Multus proxy
+watches. Service and EndpointSlice have the label, `service.kubernetes.io/service-proxy-name`, which is defined at
+[kube-proxy APIs](https://pkg.go.dev/k8s.io/kubernetes/pkg/proxy/apis), to make target service out of Kubernetes
+management.
+
+Multus proxy provides forwarding plane for multus-service with iptables at Pod's network namespace. It watches Service
+and EndpointSlice with 'service.kubernetes.io/service-proxy-name: multus-proxy' and creates changes destination to
+the service pods from Service VIP. It does not provide NAT (ip masquerade) for now because multus network is mainly
+for 'secondary network' and it assumes default route to primary Kubernetes cluster network.
+
+#### Create Service
+
+1. Service is created
+
+User creates Kubernetes service object. At that time, user needs to add following annotation/label:
+
+- `label: service.kubernetes.io/service-proxy-name` (well-known label, defined in kubernetes)
+
+Users will need to set the label `service.kubernetes.io/service-proxy-name`, with the value, `multus-proxy`. This
```

Comment:
I am curious about the naming of this. Have we thought of a naming convention that doesn't directly link it to multus? Do you think it would be a good idea to generalise the naming? i.e. additional-interface-proxy.